src/Language/Haskell/Liquid/Transforms/CoreToLogic.hs

```diff
@@ -240,48 +243,51 @@ isMeasureArg x
 varRType :: (Reftable r) => Var -> Located (RRType r)
 varRType = GM.varLocInfo ofType
 
-coreToFun :: LocSymbol -> Var -> C.CoreExpr ->  LogicM ([Var], Either Expr Expr)
-coreToFun _ _v e = go [] $ normalize e
+coreToFun :: Bool -> LocSymbol -> Var -> C.CoreExpr ->  LogicM ([Var], Either Expr Expr)
+coreToFun allowTC _ _v e = go [] $ normalize allowTC e
   where
+    isE = if allowTC then GM.isEmbeddedDictVar else isErasable
     go acc (C.Lam x e)  | isTyVar    x = go acc e
-    go acc (C.Lam x e)  | isErasable x = go acc e
+    go acc (C.Lam x e)  | isE x = go acc e
     go acc (C.Lam x e)  = go (x:acc) e
     go acc (C.Tick _ e) = go acc e
-    go acc e            = (reverse acc,) . Right <$> coreToLg e
+    go acc e            = (reverse acc,) . Right <$> coreToLg allowTC e
     
 
 instance Show C.CoreExpr where
   show = GM.showPpr
 
-coreToLogic :: C.CoreExpr -> LogicM Expr
-coreToLogic cb = coreToLg (normalize cb)
+coreToLogic :: Bool -> C.CoreExpr -> LogicM Expr
+coreToLogic allowTC cb = coreToLg allowTC (normalize allowTC cb)
 
 
-coreToLg :: C.CoreExpr -> LogicM Expr
-coreToLg (C.Let b e)
-  = subst1 <$> coreToLg e <*>  makesub b
-coreToLg (C.Tick _ e)          = coreToLg e
-coreToLg (C.App (C.Var v) e)
-  | ignoreVar v                = coreToLg e
-coreToLg (C.Var x)
+coreToLg :: Bool -> C.CoreExpr -> LogicM Expr
+coreToLg allowTC  (C.Let (C.NonRec x (C.Coercion c)) e)
```

Comment:
I believe this is added new case is an improvement regardless of whether the typeclass flag is on. Might turn out to be problematic